### DISCUSSION TITLE
Properly Loading Packages

### DISCUSSION THREAD

**Discussion by timchisamore** [ropensci/targets#313] [General]
Hello,

I am a drake user coming over to targets now that it is on CRAN and am loving it!

One question I had is regarding properly loading packages into the workflow. I know in the documentation you had mentioned using the packages argument in the tar_option_set function in the _targets.R file. However, with drake I use a requirements.R script to load packages, resolve package conflicts, set environment variables, and set the project path using the i_am function from the here package. I then would source this file in the _drake.R file before using walk from purrr to source all functions in my R directory, and finally define the drake plan. I learned this approach from @MilesMcBain [here](https://www.milesmcbain.com/posts/the-drake-post) and I believe you featured it in the drake documentation as well. I can provide further details regarding what these files look like, should that be required.

Is there a better way to do this in targets using some of the arguments to the tar_option_set function (since I'm including more than just packages)? 

All the best,

TC

**Comment by wlandau** [comment]
If you do not set the `packages` option in `tar_option_set()`, then the packages will default to whatever you load in `_targets.R`, which is equivalent to how you were using `drake`. However, if you have a lot of packages, then utilities like `tar_outdated()` and `tar_visnetwork()` will spend time loading packages they do not use. So I usually recommend `tar_option_set(packages = ...)` for whatever packages the targets need when they run.

Environment variables, packages with conflicts to manage, and `here` settings do not carry over to parallel workers created by `tar_make_clustermq()` and `tar_make_future()`. So rather than using `_targets.R` to configure all that, I recommend using `.Renviron` and `.Rprofile` files local to the project. (`.Renviron` and `.Rprofile` outside the project are ignored by `callr` and thus the `tar_make*()` functions.)

**Reply by timchisamore** [reply]
Thanks for taking the time to respond, I am going to explore this further tonight and over the next few days. I am very excited to adopt them within my workflow!
